src/components/common/CreativeCommons.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo } from 'react';
+
+import { Typography } from '@mui/material';
+
+import { DiscriminatedItem } from '@graasp/sdk';
+import {
+  CCSharingVariant,
+  CreativeCommons as GraaspCreativeCommons,
+} from '@graasp/ui';
+
+import { useLibraryTranslation } from '../../config/i18n';
+import LIBRARY from '../../langs/constants';
+
+const convertLicense = (ccLicenseAdaption: string) => {
+  // Legacy licenses.
+  if (['alike', 'allow'].includes(ccLicenseAdaption)) {
+    return {
+      requireAccreditation: true,
+      allowCommercialUse: true,
+      allowSharing: ccLicenseAdaption === 'alike' ? 'alike' : 'yes',
+    };
+  }
+
+  return {
+    requireAccreditation: ccLicenseAdaption?.includes('BY'),
+    allowCommercialUse: !ccLicenseAdaption?.includes('NC'),
+    allowSharing: (() => {
+      if (!ccLicenseAdaption || !ccLicenseAdaption.length) {
+        return '';
+      }
+      if (ccLicenseAdaption?.includes('SA')) {
+        return 'alike';
+      }
+      return ccLicenseAdaption?.includes('ND') ? 'no' : 'yes';
+    })(),
```

Comment:
Define the function above and call it below ? Here it is a bit confusing that you have the declaration and call on the same place